cedar-policy/src/ffi/utils.rs

```diff
@@ -342,6 +345,24 @@ impl Template {
             .add_template(template)
             .wrap_err(format!("failed to add template{msg} to policy set"))
     }
+
+    /// Get valid principals, actions, and resources.
+    pub fn get_valid_request_envs(
+        self,
+        s: Schema,
+    ) -> Result<(BTreeSet<String>, BTreeSet<String>, BTreeSet<String>), miette::Report> {
```

Comment:
It seems a little inconsistent to return a `HashSet<RequestEnv>` in the main API, but a `(BTreeSet<String>, BTreeSet<String>, BTreeSet<String>)` here. But maybe the interface here is the most useful when implementing an editor?

Reply:
No. It's to make return value order deterministic so that tests can pass 😂

Reply:
I think it'd be more consistent to make the APIs match

Reply:
Done.

Reply:
Doesn't seem like this is done in the current version?

To be more concrete about what I was suggesting: I think this ffi function should also return a `HashSet<RequestEnv>`.

Reply:
I changed the callee's signature to `BTreeSet` so that they are consistent. I think we should not be seeing a lot of `RequestEnv` so `BTreeSet` is fine.

Reply:
Gotcha. I think an `impl Iterator` would be more consistent with the rest of our public API than a `BTreeSet` though.

Reply:
Done.